### PR TITLE
EZP-27568: Create draft from an archived version.

### PR DESCRIPTION
**Pull request by maxbaldanza**
- [x] Create draft from an archived version.
- [x] Validate form so that only one version can be selected when creating a new draft. Any number of versions can be selected when deleting.

https://jira.ez.no/browse/EZP-27568

### CODE REVIEW

File: src/lib/Repository/UiVersionService.php [hunk 1]
```diff
@@ -30,11 +31,38 @@ public function __construct(ContentService $contentService)
...
+        $versionInfo = $this->loadVersionInfo(
+            $this->loadContentInfo($contentId),
             $versionNo
         );
 
         $this->contentService->deleteVersion($versionInfo);
     }
+
+    /**
+     * Creates a new draft based on the content and verisionNo.
```
Comment (glye) [inline src/lib/Repository/UiVersionService.php]: versionNo

File: src/lib/Repository/UiVersionService.php [hunk 2]
```diff
@@ -30,11 +31,38 @@ public function __construct(ContentService $contentService)
...
+     *
+     * @param mixed $contentId
+     * @param mixed $versionNo
+     */
+    public function createDraft($contentId, $versionNo)
+    {
+        $contentInfo = $this->loadContentInfo($contentId);
+        $versionInfo = $this->loadVersionInfo(
+            $this->loadContentInfo($contentId),
```
Comment (bdunogier) [inline src/lib/Repository/UiVersionService.php]: You can use `$contentInfo` from above. And I don't think it's worth wrapping that line then.

File: src/lib/Repository/UiVersionService.php (hunk 1 above)
Comment (bdunogier) [inline src/lib/Repository/UiVersionService.php]: Typo: `versionNo`.

File: src/lib/Repository/UiVersionService.php [hunk 3]
```diff
@@ -30,11 +31,38 @@ public function __construct(ContentService $contentService)
...
             $versionNo
         );
 
         $this->contentService->deleteVersion($versionInfo);
     }
+
+    /**
+     * Creates a new draft based on the content and versionNo.
```
Comment (bdunogier) [inline src/lib/Repository/UiVersionService.php]: `contentId`, actually.

File: src/bundle/Controller/VersionController.php [hunk 4]
```diff
@@ -58,6 +58,7 @@ public function archiveActionsAction(
 
         if ($archiveActionsForm->isValid()) {
             $this->deleteVersionsBasedOnFormSubmit($archiveActionsForm, $content->id);
+            $this->createDraftVersionBasedOnFormSubmit($archiveActionsForm, $content->id);
```
Comment (bdunogier) [inline src/bundle/Controller/VersionController.php]: Given that you have a `Content`, is it really worth passing the ID ? It is gonna be loaded back to a `ContentInfo` later on. You might as well pass the `ContentInfo` directly.

**Review by mcurran-inviqa — APPROVED** [review]

**Review by bdunogier — CHANGES_REQUESTED** [review]
I'm overall a bit unsure about falling back to id as arguments.

Our Repository's API is on purpose mostly manipulating Value Objects, and I think that we should, when we can, use them instead of scalards in this layer as well.

File: src/bundle/Form/Versions/ArchivedActions.php [hunk 5]
```diff
@@ -15,7 +18,27 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         parent::buildForm($builder, $options);
 
         $builder
-            ->add('delete', SubmitType::class)
+            ->add('delete', SubmitType::class, array(
+                'validation_groups' => false,
```
Comment (mcurran-inviqa) [inline src/bundle/Form/Versions/ArchivedActions.php]: use [] syntax for consistency

**Review by mcurran-inviqa — APPROVED** [review]

**Review by bdunogier — APPROVED** [review]

**Pull request merged**